module_3/src/com/epam/stringasstringorstringbuilder/StringModifier.java

```diff
@@ -0,0 +1,43 @@
+package com.epam.stringasstringorstringbuilder;
+
+public class StringModifier {
+
+    // task_02
+    public static String insertTextAfterSpecifiedSubstring(String str, String textToInsert, String specifiedSubstr) {
+        String resultString = "";
+        if (str.contains(specifiedSubstr)) {
+            resultString = str.replace(specifiedSubstr, specifiedSubstr + textToInsert);
+        }
+        return resultString;
+    }
+
+    // task_04
+    public static String concatenateSpecifiedWordFromAnotherWord(String donorWord, String specifiedWord) {
+        String resultStr = "";
+        boolean thereAreNotAllCharacters = false;
+        int index;
+        for (int i = 0; i < specifiedWord.length(); i++) {
+            index = donorWord.indexOf(specifiedWord.charAt(i));
```

Comment:
1. You need index only in cycle, so define it there
2. The same comment about specifiedWord.length() as in 1st task
```suggestion
        final int AAA = specifiedWord.length();
        for (int i = 0; i < AAA; i++) {
            int indexOfSearchingLetter= donorWord.indexOf(specifiedWord.charAt(i));
```